Docstring collection failure with ellipsis

Hi, was just updating to the new version of sybil v7.1.0 (thanks, its very useful!) and I ran into a bug with using ellipses here is a minimal example:

A test file:
```python
class A: ...

def test_hello():
   assert True
```

And conftest.py:
```python
import doctest

from sybil import Sybil
from sybil.parsers.rest import (
    DocTestParser,
    PythonCodeBlockParser,
)

rest_examples = Sybil(
    parsers=[
        DocTestParser(doctest.ELLIPSIS | doctest.NORMALIZE_WHITESPACE),
        PythonCodeBlockParser(),
    ],
    patterns=["*.py", "*.rst"],
    excludes=["index.rst"],  
    fixtures=["tmp_path"],
)

pytest_collect_file = rest_examples.pytest()
```

with this setup and just running `pytest`  I run into an attribute error because the regex doesnt find any docstring on the class A (or any code I guess) although it is valid python:
```bash
________ ERROR collecting test.py ____________________________________________________
<clipped>/lib/python3.11/site-packages/sybil/integration/pytest.py:109: in collect
    document = sybil.parse(self.path)
<clipped>/lib/python3.11/site-packages/sybil/sybil.py:152: in parse
    return type_.parse(str(path), *self.parsers, encoding=self.encoding)
<clipped>//lib/python3.11/site-packages/sybil/document.py:246: in parse
    for start, end, text in cls.extract_docstrings(document.text):
<clipped>//lib/python3.11/site-packages/sybil/document.py:233: in extract_docstrings
    punc_size = len(punc.group(1))
E   AttributeError: 'NoneType' object has no attribute 'group'
```

--- a/setup.cfg
+++ b/setup.cfg
@@ -14,3 +14,5 @@ omit =
 exclude_lines =
     pragma: no cover
     if TYPE_CHECKING:
+    \.\.\.
+

--- a/sybil/document.py
+++ b/sybil/document.py
@@ -225,6 +225,8 @@ class PythonDocStringDocument(PythonDocument):
                 text = docstring.value
             else:
                 continue
+            if text is Ellipsis:
+                continue
             node_start = line_offsets.get(docstring.lineno-1, docstring.col_offset)
             end_lineno = docstring.end_lineno or 1
             end_col_offset = docstring.end_col_offset or 0

--- a/tests/samples/docstrings.py
+++ b/tests/samples/docstrings.py
@@ -45,3 +45,7 @@ def decorator(fn):
     ...     \"\"\"docstring of a decorated function\"\"\"
 
     """
+
+
+def ellipsis_as_body():
+    ...

